### DISCUSSION TITLE
Agent lost link to server

### DISCUSSION THREAD

**Discussion by orugari** [amidaware/tacticalrmm#1336] [General]
**Server Info (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - RMM Version (as shown in top left of web UI): 0.15.2

**Installation Method:**
  - [ ] Standard

**Agent Info (please complete the following information):**
- Agent version (as shown in the 'Summary' tab of the agent from web UI): 2.4.0
- Agent OS: Win 10

**Describe the bug**
I have few agent that appear always offline. I tried to do an Agent recover but no success. I finally installed again the agent and a dupplicate WorkStation appeard. I didn't change anything on the client computer (name, domain or whatever).
It looks like also on some computer, TRMM completely dissapear. No antivirus blocking it, users has no access to uninstall feature or delete from Program Files.

**To Reproduce**
1. Agent appear offline on TRMM, but conputer is online

**Expected behavior**
Response from Agent

**Screenshots**
![Capture d’écran 2022-10-28 101124](https://user-images.githubusercontent.com/4814790/198538182-fdd5bfb5-7cb4-4896-852e-5b86d9c9ea1a.png)

**Comment by dinger1986** [comment]
this looks like an issue with your setup, if you reinstall a double will always show in the web gui.

If this isnt the case then is anything connected on the same site and staying connected? 

You havent given any info about your setup ie if its running behind a proxy or whatever.

The agent thats offline is showing that it needs updates, is that the one thats stuck on 2.4.0?

**Reply by orugari** [reply]
Probably an update failed then, because no problem occured from an other computer on the same network, and especially if the TacticalRMM folder is not anymore present.
The offline is on 2.4.0 yes, the other I did reinstall it.

**Reply by dinger1986** [reply]
so within Program Files TacticalRMM folder had disappeared? If thats the case something has removed it, updates dont remove the folder as its only updating the agent in the folder and the meshagent at a minimum should still be in there. What AV do you have?

**Reply by orugari** [reply]
Sophos. but I made rule to prevent any quarantine or whatsoever 
I just checked an other computer, same problem (not same person or group of person), no more folder and I noticed a failed update to the last version. I have other cases like that, I try to get log or any hint about that situation. If I found something (I'm 99.99% sure it's not someone that deleted it manually) I'll report it.

**Reply by dinger1986** [reply]
Seen it with Sophos even with rules doing funny things.
